lib/medium-admin.php

```diff
@@ -1097,7 +1102,10 @@ private static function _handle_response($response) {
       $error_code = $response->get_error_code();
       error_log("WP ERROR: $message ($error_code)");
       if ($error_code == "http_request_failed" && strpos($message, "timed out") !== false) {
-        throw new Exception($message, -2); // our custom code for timeouts
+        throw new Exception($message, API_TIMEOUT);
+      }
+      if ($error_code == "http_request_failed" && strpos($message, "provisioned throughput for the table was exceeded") !== false) {
```

Comment:
What - do we leak these errors all the way out to 3rd party clients? That doesn't seem like a great idea :/

(I consider WP plugin in the same class as 3rd party clients as it uses same APIs)